recipes-openxt/tss/tpm2-tools/tpm2-tools-lib-support.patch

```diff
@@ -270,111 +321,123 @@ index 4a998d2..caf976f 100644
 +#define TEST_PCR_SELECT_BIT( pcrSelection, pcr ) \
 +    ((pcrSelection).pcrSelect[( (pcr)/8 )] & ( 1 << ( (pcr) % 8) ))
 +
- 
  #endif /* SRC_PCR_H_ */
+diff --git a/lib/tpm2_util.c b/lib/tpm2_util.c
+index 57d6c76..ca1230a 100644
+--- a/lib/tpm2_util.c
++++ b/lib/tpm2_util.c
+@@ -106,6 +106,18 @@ bool tpm2_util_string_to_uint32(const char *str, uint32_t *value) {
+     return true;
+ }
+ 
++int tpm2_util_string_to_byte_structure(const char *inStr, UINT16 *byteLength, BYTE *byteBuffer) {
++    if(inStr == NULL || byteLength == NULL || byteBuffer == NULL)
++        return -1;
++    if(*byteLength <= strlen(inStr))
++        return -2;
++
++    *byteLength = strlen(inStr);
++    memcpy(byteBuffer, inStr, *byteLength);
++    byteBuffer[*byteLength] = '\0';
++    return 0;
++}
```

Comment:
I find the naming on that one a little confusing, especially since the previous defined function, `tpm2_util_string_to_uint32`, converts a string input to an unsigned int.
This one is just takes a raw buffer with a nul terminator and copies it to the `

Also, the input from this seems to be from command-line (`tpm2_sealdata -K` and `tpm2_unsealdata -P`), where hex values are easier to handle. With `tpm2_util_hex_to_byte_structure` already there, shouldn't the call sites be changed so this function is no longer required?

Reply:
Didn't catch this comment on my first read through. This function was actually one of the first things I changed in this endeavor; now that I know more about the API I can likely remove it.